### PR TITLE
ENT-4156 | Update taxonomy models to pull skill tag descriptions

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
added description field in Skill model and update the refresh_course_skill command to save skill description.

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [x] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/utils.py [hunk 1]
```diff
@@ -11,11 +11,11 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def update_skills_data(course_key, confidence, skill_data):
+def update_skills_data(course_key, skill_external_id, confidence, skill_data):
     """
     Persist the skills data in the database.
     """
-    skill, __ = Skill.objects.update_or_create(**skill_data)
+    skill, __ = Skill.objects.update_or_create(external_id=skill_external_id, defaults=skill_data)
```
Comment (HammadAhmadWaqas) [inline taxonomy/utils.py:18]: @muhammad-ammar @saleem-latif made changes here as well. please review this part with extra care.

File: taxonomy/emsi_client.py [hunk 2]
```diff
@@ -153,6 +153,15 @@ def traverse_data(response):
         """
         Transform data to a more useful format.
         """
+        for skill_details in response['data']:
+            # append skill description in skill data extracted from "wikipediaExtract" tag
+            try:
+                desc = next(tag['value'] for tag in skill_details['skill']['tags'] if tag['key'] == 'wikipediaExtract')
+            except StopIteration:
+                LOGGER.warning('[TAXONOMY] "wikipediaExtract" key not found in skill: %s', skill_details['skill']['id'])
+                desc = ''
```
Comment (muhammad-ammar) [inline taxonomy/emsi_client.py:162]: I think we can simplify this whole block. Looks like next can take a default value which will be returned if we nothing is found. next docs https://docs.python.org/3/library/functions.html#next

Comment (HammadAhmadWaqas) [reply taxonomy/emsi_client.py:162]: @muhammad-ammar Yes you are right, I did it this way so that I could log a warning as well.

Comment (muhammad-ammar) [reply taxonomy/emsi_client.py:162]: @HammadAhmadWaqas Is there a possibility that `wikipediaExtract` will not be present?

**Review by saleem-latif — APPROVED** [review]

File: taxonomy/emsi_client.py (hunk 2 above)
Comment (HammadAhmadWaqas) [reply taxonomy/emsi_client.py:162]: `wikipediaExtract` is just a tag key, It is not mentioned in the docs it's a mandatory tag or not.
Added logs here to assure that if there will be missing description from our system, we will be able to confirm that is it due to the fact the API is not returning `WikipediaExtract`

**Review by muhammad-ammar — APPROVED** [review]

**Pull request merged**